Dev

MrDarkSpace opened the pull request:
Следует просто запустить tamagochi.py в любом интерпретаторе.

Описание:
Код реализует процесс игры в Тамагочи. В начале игра предложит выбрать животное, она дождется, пока не будет верно введен вид животного. Далее она попросит ввести имя.
После выполнения предыдущих пунктов будут выведены характеристики животного.
Животное имеет 4 основные характеристики:
1. Возраст (возрастает на 1 год каждые 30 секунд)
2. Еда (падает на 1 каждые 3 секунды; если поднимается выше 95 - происходит поход в туалет + создание одного "Неубранного похода в туалет")
3. Чистота (падает падает на 1*"Неубранные походы в туалет" каждые 3 секунды)
4. Здоровье (падает на 1 каждые 10 секунд; Начинает падать быстрее, если Еда или Чистота ниже 30)

Доступные команды:
1. Feed - покормить. Предлагает выбрать порцию еды. Нельзя поднять еду выше 95.
2. Wash - +50 к Чистоте
3. Cure - +30 к Здоровью
4. Clean - уборка туалета. Убирает один "Неубранный поход в туалет"

После ввода любой команды печатается настроение и состояние питомца.
Ввод команд не чувствителен к регистру. Если команда не распознана, просто печатается состояние питомца.
Если любая из характеристик падает до 0, то питомец умирает.

MrDarkSpace committed:
Add code

MrDarkSpace committed:
Add some comments to code

MrDarkSpace committed:
Some changes in plus_age

febos commented on tamagochi.py:
```diff
@@ -0,0 +1,166 @@
+    cls.Cleanness = cleanness                                 #Чистота
+    cls.Health = health                                       #Здоровье
+    cls.Age = age                                             #Возраст
+    return super().__new__(cls)
```
тут происходит инициализация объекта, хотя для этого предназначен метод __init__ - ваш питомец всё-таки объект, а не класс

febos commented on tamagochi.py:
```diff
@@ -0,0 +1,166 @@
+    if (self.Food <= 30) or (self.Cleanness <= 30):
+      self.Health = self.Health - int((time.time()-st)//3)
+    else:
+      self.Health = self.Health - int((time.time() - st) // 10)
```
избегайте лишних повторений в коде

febos commented on tamagochi.py:
```diff
@@ -0,0 +1,166 @@
+      self.Health = self.Health - int((time.time() - st) // 10)
+
+  def feed(self):    #Процесс Кормления
+    feedtype = str(input('Choose serving size: \'1\' (+5 Food), \'2\' (+10 Food) or \'3\' (+20 Food).\n'))
```
избавьтесь от длинных строк (PEP-8)

febos commented on tamagochi.py:
```diff
@@ -0,0 +1,166 @@
+  def feed(self):    #Процесс Кормления
+    feedtype = str(input('Choose serving size: \'1\' (+5 Food), \'2\' (+10 Food) or \'3\' (+20 Food).\n'))
+    while True:
+      if (feedtype == '1') or (feedtype == '2') or (feedtype == '3'):
```
для этого есть in

febos commented on tamagochi.py:
```diff
@@ -0,0 +1,166 @@
+    feedtype = str(input('Choose serving size: \'1\' (+5 Food), \'2\' (+10 Food) or \'3\' (+20 Food).\n'))
+    while True:
+      if (feedtype == '1') or (feedtype == '2') or (feedtype == '3'):
+        self.Food += 5*int(feedtype) + (int(feedtype)//3)*5
```
используйте константы вместо голых значений

febos commented on tamagochi.py:
```diff
@@ -0,0 +1,166 @@
+      break
+    elif (animaltype == 'g'):
+      animal = Goat()
+      break
```
это удобно сделать словарем

febos commented on tamagochi.py:
```diff
@@ -0,0 +1,166 @@
+
+  animal.name = input('Now you should name your pet. Choose wisely.\n')
+  print('Wow! Great name: ' + animal.name + '.\nSo, now we are starting our game! Be very attentive to your ' +
+        animal.name + '!')
```
длинные строки

febos commented on tamagochi.py:
```diff
@@ -0,0 +1,166 @@
+
+    start_time = time.time()
+
+    if (animal.Food > 80) and (animal.Cleanness > 80) and (animal.Health > 80):
```
для этого есть функция all

febos commented on tamagochi.py:
```diff
@@ -0,0 +1,166 @@
+      animal.toilet()
+      animal.Food = 95
+
+    feedback = animal.face + animal.name + ' is ' + animal.state + " 💩"*animal.toi_number + "\n" +\
```
смайлик в коде - не тру - используйте его юникод-номер

febos reviewed (commented):
1) Если сначала подождать хотя бы 30 секунд перед первой командой, то потом можно много раз быстро нажимать ввод и тогда возраст с каждым нажатием растет на фиксированное значение (зависит от времени до первой команды), а остальные показатели не меняются совсем, независимо от продолжительности повторных нажатий

2) см. комментарии к коду

3) У вас неделя на поправки

MrDarkSpace committed:
Solving some mistakes

Solved:
1. Problem of food_drop, clnns_drop, health_drop (didn't drop if you are clicking very freaquently)
2. Problem of plus_age (strange behavior)
3. mistake with __new__ and __init__
4. too long lines
and some other

febos commented:
проект принят
